unittests/savanna_misc_tests.cpp

```diff
@@ -681,10 +799,30 @@ BOOST_FIXTURE_TEST_CASE(verify_spring_1_0_block_compatibitity, savanna_cluster::
    BOOST_REQUIRE_EQUAL(qc_s(qc(b9)), strong_qc(b8));    // b9 claims a strong QC on b8
    BOOST_REQUIRE_EQUAL(A.lib_number, b6->block_num());  // b9 makes B6 final
 
-   // check that the block id of b9 match what we got with Spring 1.0.0
+   std::filesystem::path test_data_path { UNITTEST_TEST_DATA_DIR };
+   auto ref_blockchain_path = test_data_path / "consensus_blockchain";
+
+   // check that the block id of b9 match what we got before.
    auto b9_id = b9->calculate_id();
-   BOOST_REQUIRE_EQUAL(b9_id, block_id_type{"00000013725f3d79bd4dd4091d0853d010a320f95240981711a942673ad87918"});
+   BOOST_REQUIRE_EQUAL(b9_id, read_reference_id(ref_blockchain_path));
+
+   if (save_blockchain) {
+      save_blockchain_data(ref_blockchain_path, A.get_config().blocks_dir, b9_id, snapshot);
+      return;
+   }
```

Comment:
I believe this should be before line 807, otherwise is there is a need to save the blockchain (meaning that it is different from what is already saved), line 807 would fail first.